compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SyntheticAccessorLowering.kt

```diff
@@ -443,21 +445,60 @@ internal class SyntheticAccessorLowering(val context: JvmBackendContext) : IrEle
     }
 
     // Counter used to disambiguate accessors to functions with the same base name.
-    private var nameCounter = 0
+    private val nameCounterMap = mutableMapOf<IrFunction, Int>()
+
+    private fun drawNextNameSuffix(function: IrFunction): Int {
+        val next = nameCounterMap[function] ?: 0
+        nameCounterMap[function] = next + 1
+        return next
+    }
 
-    private fun IrFunction.accessorName(): Name {
+    private fun IrFunction.accessorName(superQualifier: IrClassSymbol?): Name {
         val jvmName = context.methodSignatureMapper.mapFunctionName(this)
-        return Name.identifier("access\$$jvmName\$${nameCounter++}")
+        val suffix = when {
+            // The only function accessors placed on interfaces are for JvmDefault implementations
```

Comment:
Actually there could be private JvmDefault function than we need accessor for it. But we still can use $jd suffix cause there would no any accessor for it in compatibility mode. Please mention it in comment

Reply:
Or we can just check private visibility here and use default suffix in such case

Reply:
Please also add test for such case

Reply:
You are thinking something like the following?
```
interface I {
  @JvmDefault
  private fun foo() = 4

  fun bar() = {foo() + 5}()
}
```
The old and new backend actually differ on where to place the accessor. The old backend accesses `foo` via `I`, so the accessor goes there. IR goes via `I$DefaultImpls`. Might it be due to inlining of lambdas?

> Or we can just check private visibility here and use default suffix in such case

When you say "default" you mean the sequence number, "s+hashcode" or do you mean "jd"?

Reply:
It should be same accessor as in 
```
class I {

  private fun foo() = 4

  fun bar() = {foo() + 5}()
}
```